### PR TITLE
Serve appropriate assessment event.

### PR DESCRIPTION
**Pull request by lukaDoric**

### CODE REVIEW

File: src/app/modules/domain/knowledge-component/knowledge-component.component.ts [hunk 1]
```diff
@@ -54,9 +57,10 @@ export class KnowledgeComponentComponent implements OnInit {
     });
   }
 
-  private getAssessmentEvents(): void {
-    this.unitService.getAssessmentEvents(this.kcId).subscribe(assessmentEvents => {
-      this.learningObjects = assessmentEvents;
+  private getSuitableAssessmentEvent(): void {
+    this.unitService.getSuitableAssessmentEvent(this.kcId, this.learnerId).subscribe(assessmentEvent => {
+      this.learningObjects = [];
```
Comment (lukaDoric) [inline src/app/modules/domain/knowledge-component/knowledge-component.component.ts:62]: This is the way i set learningObjects because now we have only one AE.

File: src/app/modules/domain/knowledge-component/knowledge-component.component.html [hunk 2]
```diff
@@ -26,9 +26,9 @@ <h2 class="mat-h2">
       </button>
       <h3>Goal: {{knowledgeComponent.name}}</h3>
       <mat-divider></mat-divider>
-      <cc-learning-object-container
-        *ngFor="let learningObject of learningObjects"
-        [learningObject]="learningObject"></cc-learning-object-container>
+        <cc-learning-object-container
```
Comment (Luburic) [inline src/app/modules/domain/knowledge-component/knowledge-component.component.html:29]: This seems to be a mistake? Revert it if it is the case.

File: src/app/infrastructure/app-routing.module.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ const routes: Routes = [
   {path: 'login', component: LoginComponent},
   {path: 'register', component: RegisterComponent},
   {path: 'unit/:unitId', component: UnitComponent, resolve : {unit: UnitResolver}},
-  {path: 'kc/:kcId', component: KnowledgeComponentComponent, canActivate: [AuthGuard]},
+  {path: 'kc/:kcId/learner/:learnerId', component: KnowledgeComponentComponent, canActivate: [AuthGuard]},
```
Comment (Luburic) [inline src/app/infrastructure/app-routing.module.ts:16]: We can't store the user Id in the URL. URLs are supposed to target specific resources. The user's id is an application-wide piece of data, similar to the selected theme. We already have a service for caching it. See https://github.com/Clean-CaDET/platform-tutor-ui-web/blob/master/src/app/modules/domain/feedback/feedback.service.ts#L21 for an example.

File: src/app/modules/layout/navbar/navbar.component.html [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
         {{kc.name}}
       </button>
-    </mat-menu>
-    <button color="primary" *ngIf="learner" id="profile" mat-flat-button [routerLink]="['profile']">Profile</button>
-    <button color="primary" *ngIf="!learner" id="login" mat-flat-button [routerLink]="['login']">Login</button>
-    <button color="primary" *ngIf="learner" mat-flat-button id="logout" (click)="onLogout()" [routerLink]="['home']">Logout</button>
+    </ng-container>
+  </mat-menu>
+  <button color="primary" *ngIf="learner" id="profile" mat-flat-button [routerLink]="['profile']">Profile</button>
+  <button color="primary" *ngIf="!learner" id="login" mat-flat-button [routerLink]="['login']">Login</button>
```
Comment (Luburic) [inline src/app/modules/layout/navbar/navbar.component.html:28]: Are these changes only stylistic and enforced by your IDE? If so, we should setup a common convention to avoid always overriding each other's styles.

Please send me your configuration if you've already established some conventions following best practices. If not, revert the changes and disable any automatic styling while we decide on the common set of conventions in the coming months.

**Review by Luburic — CHANGES_REQUESTED** [review]
Comments inline.

File: src/app/modules/layout/navbar/navbar.component.html (hunk 4 above)
Comment (lukaDoric) [reply src/app/modules/layout/navbar/navbar.component.html:28]: No, there was problem with rebase. There were two files under name ''navbar.component.html" so i removed one and moved other to modules/layout/navbar/.

Comment (lukaDoric) [reply src/app/modules/layout/navbar/navbar.component.html:23]: Reverted.

**Review by Luburic — APPROVED** [review]
